syftlib/src/test/java/org/openmined/syft/unit/monitor/network/NetworkStatusRealDataSourceTest.kt

```diff
@@ -0,0 +1,155 @@
+package org.openmined.syft.unit.monitor.network
+
+import android.accounts.NetworkErrorException
+import android.net.NetworkCapabilities
+import org.junit.Test
+import org.openmined.syft.common.AbstractSyftWorkerTest
+import org.openmined.syft.common.HttpClientMock
+import org.openmined.syft.common.SocketClientMock
+import org.openmined.syft.domain.SyftConfiguration
+import org.openmined.syft.monitor.network.NetworkStatusModel
+import org.openmined.syft.monitor.network.NetworkStatusRealTimeDataSource
+
+@ExperimentalUnsignedTypes
+class NetworkStatusRealDataSourceTest : AbstractSyftWorkerTest() {
```

Comment:
Does this class need abstract worker for all the tests? Abstract syft worker uses robolectric which makes tests slower. If possible use it only when you absolutely require the context (it cannot be mocked)

Reply:
The context is required in initializing NetworkStatusRealTimeDataSource

Reply:
Yeah but a mock Context can be supplied there unless we need interactions with context that can't be stubbed like setting up network validity constraints (so except for network validity I don't see the need of context and it can simply be mocked)